[Contextual Security] Add support to show generic entity popover when expanding entity details

alexreal1314 opened the pull request:
## Summary

Purpose of this PR is to add the capability of showing generic entity popover when clicking on 'Show entity details' on entities in the graph. If an entity doesn't have the necessary data - the option in the menu is disabled.

- add isPreviewMode support to GenericEntityPanel with conditional footer rendering
- update Enhance entity node popover logic
- update Storybook examples to demonstrate disabled state functionality
 -update fetchGraph, parseRecords, generic entity panel tests


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [x] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

Since the generic entity flyout requires the document_id fields which we currently don't have
There are a few options to solve the issue of showing the generic entity popover from the graph view:

1. Expose `document _id and _index meta fields via the entities-v1-latest-security_generic` ingest pipeline, for example as entity.metadata._id and entity.metadata._index.
After doing so we expose them also via the enrich policy, e.g `entity_store_field_retention_generic_<space>_v1.0.0` which is the one we are using today. That way the _id and _index fields of the ingested entity is exposed and could be used to show the generic entity flyout.

Why this option was not chosen?

requires changes in the entity store infra - might introduce issues we are not aware of.

2. Make a separate query with the entity.id field before opening the generic entity flyout to fetch the document _id field and than passing it to the flyout.

Why this option was not chosen?

Makes us make another query thus there is a loading phase which is handled outside the scope of the flyout - seems wrong in terms of user experience and might introduce some latency.

3. Make the generic entity flyout component and hook to accept either doc _id or entity.id. There are no breaking changes as we allow to pass either of them or both. If both are passed we query for doc _id, otherwise we query the term which is passed. Entity.id is a unique identifier therefor it is a valid approach.


**Option 3 is implemented in this PR.**

### Screenshots

Entity popover open - when node is enriched with entity data:
<img width="1645" height="933" alt="image" src="https://github.com/user-attachments/assets/b4279f3d-2f80-48be-82ed-58f214e450bd" />

Show entity details is disabled - when the node has no corresponding entity data:
<img width="1655" height="934" alt="image" src="https://github.com/user-attachments/assets/ed3fa886-87ee-4a5e-bd17-ffb4844ec4cb" />

graph without entity data/entity store not enabled/generic entity enrich policy doesn't exist:
<img width="1655" height="932" alt="image" src="https://github.com/user-attachments/assets/44c0a7be-133c-46a1-9cfb-f78090f49930" />

<img width="1633" height="933" alt="image" src="https://github.com/user-attachments/assets/37f6c418-c5a7-4994-8076-74c14f7cbd21" />

alexreal1314 committed:
extract doc_id and doc_index from enrich policies for documentsData and generic entity popover

add isPreviewMode support to GenericEntityPanel with conditional footer rendering

update Enhance entity node popover logic

update Storybook examples to demonstrate disabled state functionality

update fetchGraph, parseRecords, generic entity panel and graph invetigation storybook tests

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/328604)
* Commit: 49d57d569efe8b4449cdb209f8df87a95ce7be89
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-231370)

### Failed CI Steps
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/328604#01989d50-0800-43e0-87a8-117f77354bbd)
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/328604#01989d7c-7ba1-4eeb-b847-664e651804b1)
* [FTR Configs #83](https://buildkite.com/elastic/kibana-pull-request/builds/328604#01989d50-0816-45d3-98f2-3eeaa520610e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328604#01989d50-0816-45d3-98f2-3eeaa520610e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328604/jobs/01989d50-0816-45d3-98f2-3eeaa520610e/artifacts/01989d60-1252-421f-b9fd-5dc756746cd9) FTR Configs #<span></span>83 / alerting api integration security and spaces enabled - Group 2 Alerting and Actions Telemetry test telemetry should retrieve telemetry data in the expected format
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328604#01989d50-0800-43e0-87a8-117f77354bbd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328604/jobs/01989d50-0800-43e0-87a8-117f77354bbd/artifacts/01989d7c-3f6b-435c-843b-2a4f65167d67) FTR Configs #<span></span>57 / Cloud Security Posture POST /internal/cloud_security_posture/graph Happy flows Enrich graph with entity metadata should contain entity data when asset inventory is enabled
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328604#01989d7c-7ba1-4eeb-b847-664e651804b1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328604/jobs/01989d7c-7ba1-4eeb-b847-664e651804b1/artifacts/01989d84-253e-4f3f-9a23-728e5e16983f) FTR Configs #<span></span>57 / Cloud Security Posture POST /internal/cloud_security_posture/graph Happy flows Enrich graph with entity metadata should contain entity data when asset inventory is enabled

### History
* :broken_heart: [Build #328550](https://buildkite.com/elastic/kibana-pull-request/builds/328550) failed 1f873a047add06af221e6bbfa3a87d6c1afe5eaf

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328604","number":328604,"commit":"49d57d569efe8b4449cdb209f8df87a95ce7be89"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328550","number":328550,"commit":"1f873a047add06af221e6bbfa3a87d6c1afe5eaf"}],"number":328604}
buildkite-pr-comment-->

alexreal1314 committed:
remove duplications in graph integration tests and fix fetch graph mock data

alexreal1314 committed:
update useGetGenericEntity to accept both entityDocId and entityId parameters

remove sourceDocId and index fields from entity data structure

update graph visualization entity preview handling

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/328889)
* Commit: cb11233c032a348b250a9885325e9059846d193c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-231370)

### Failed CI Steps
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/328889#01989fd6-4c16-49fd-9627-90b0812950a3)
* [Jest Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/328889#01989fd6-4c18-4376-9881-87428c160fa1)
* [Jest Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/328889#01989ffb-aa97-4d27-bf0c-7a7a2f8cce59)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328889#01989fd6-4c16-49fd-9627-90b0812950a3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328889/jobs/01989fd6-4c16-49fd-9627-90b0812950a3/artifacts/0198a002-3dd7-424f-86a6-19e2c1caf8ce) Jest Tests #<span></span>2 / fetchGenericEntity query building logic should treat whitespace-only docId as empty and use entityId instead
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328889#01989fd6-4c18-4376-9881-87428c160fa1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328889/jobs/01989fd6-4c18-4376-9881-87428c160fa1/artifacts/01989ffb-8573-4a2a-98eb-8e1f8d3a9749) Jest Tests #<span></span>7 / GraphInvestigation Component popover shows `Show entity details` as disabled when entity node has documentsData with no sourceDocId
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/328889#01989ffb-aa97-4d27-bf0c-7a7a2f8cce59) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/328889/jobs/01989ffb-aa97-4d27-bf0c-7a7a2f8cce59/artifacts/0198a002-afc2-436f-b6df-db11aaccd721) Jest Tests #<span></span>7 / GraphInvestigation Component popover shows `Show entity details` as disabled when entity node has documentsData with no sourceDocId

### History
* :broken_heart: [Build #328604](https://buildkite.com/elastic/kibana-pull-request/builds/328604) failed 49d57d569efe8b4449cdb209f8df87a95ce7be89
* :broken_heart: [Build #328550](https://buildkite.com/elastic/kibana-pull-request/builds/328550) failed 1f873a047add06af221e6bbfa3a87d6c1afe5eaf

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328889","number":328889,"commit":"cb11233c032a348b250a9885325e9059846d193c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328604","number":328604,"commit":"49d57d569efe8b4449cdb209f8df87a95ce7be89"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328550","number":328550,"commit":"1f873a047add06af221e6bbfa3a87d6c1afe5eaf"}],"number":328889}
buildkite-pr-comment-->

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
Cloud deployments require a Github label, please add `ci:cloud-deploy` or `ci:cloud-redeploy` and trigger the job through the checkbox again.

alexreal1314 committed:
fix useGetGenericEntity params types to accept either doc _id or entity.id fields

refactor generic entity panel related types

alexreal1314 commented on x-pack/solutions/security/plugins/security_solution/public/flyout/entity_details/generic_details_left/index.tsx:
```diff
@@ -70,15 +74,18 @@ const useSelectedTab = (params: GenericEntityDetailsPanelProps, tabs: LeftPanelT
+  // By passing the entire params object, we maintain the union type constraints that enforce
+  // either entityDocId or entityId to be present
+  // Destructuring of the params to extract the relevant fields happens internally in useGetGenericEntity
+  const { getGenericEntity } = useGetGenericEntity(params);
```
we can't pass here:
`const { getGenericEntity } = useGetGenericEntity({entityDocId, entityId});
`
Because we get compilation errors from typescript, as it doesn't infer well union types.
<img width="764" height="216" alt="image" src="https://github.com/user-attachments/assets/fd0e0a51-d103-4ac1-93f9-4ef8d0073c04" />

albertoblaz replied:
The reason is, if you destructure, TS infers you'd be passing in a "generic" object where both fields might be optional strings (undefined). So you need to type-cast the object as a `UseGetGenericEntityParams` type. This narrows it down to the type the function expects and removes the error:

```ts
const { getGenericEntity } = useGetGenericEntity({
  entityDocId,
  entityId,
} as UseGetGenericEntityParams);
```

alexreal1314 commented on x-pack/solutions/security/test/cloud_security_posture_api/routes/graph.ts:
```diff
@@ -1049,174 +1068,6 @@ export default function (providerContext: FtrProviderContext) {
           expect(edge.type).equal('solid');
         });
       });
-
```
some duplication of tests was added in one of the previous PRs.

albertoblaz commented on x-pack/solutions/security/plugins/security_solution/public/flyout/entity_details/generic_right/hooks/use_get_generic_entity.ts:
```diff
@@ -17,28 +17,49 @@ import { useKibana } from '../../../../common/lib/kibana';
 
-const fetchGenericEntity = async (
+// explicitly force to pass either parameter or both
+export type UseGetGenericEntityParams =
+  | { entityDocId: string; entityId?: never }
+  | { entityDocId?: never; entityId: string }
+  | { entityDocId: string; entityId: string };
```
Very nice!

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/graph_investigation.stories.test.tsx:
```diff
@@ -204,21 +204,23 @@ describe('GraphInvestigation Component', () => {
       const showDetailsItem = queryByTestId(GRAPH_NODE_POPOVER_SHOW_ENTITY_DETAILS_ITEM_ID);
-      expect(showDetailsItem).not.toBeInTheDocument();
+      expect(showDetailsItem).toHaveTextContent('Show entity details');
+      expect(showDetailsItem).toHaveAttribute('disabled');
```
@albertoblaz thanks, test case title are wrong, will fix it.

alexreal1314 committed:
fix graph investigation test case titles

albertoblaz reviewed (approved):
Looks pretty good and I see no reason to block this, so I approve the changes :)

However, I would still check with @kfirpeled since he has a broader picture of everything.

hop-dev reviewed (approved):
Entity analytics changes look good thanks

kfirpeled commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/common/schema/graph/v1.ts:
```diff
@@ -56,8 +57,13 @@ export const entitySchema = schema.object({
+    schema.literal(DOCUMENT_TYPE_SINGLE_ENTITY),
+  ]),
   index: schema.maybe(schema.string()),
+  sourceDocId: schema.maybe(schema.string()),
```
this is not in use

is it redundant?

kfirpeled reviewed (commented):
few comments, I'm waiting for serverless image to be built to test it further

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/330216)
* Commit: ff25739b8a75df9c00b3e8b3fa21b265ef87a165
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-231370-ff25739b8a75`

### History
* :yellow_heart: [Build #328996](https://buildkite.com/elastic/kibana-pull-request/builds/328996) was flaky 3d135c12d3ce43fdfc588d0d488fe650770062db
* :green_heart: [Build #328943](https://buildkite.com/elastic/kibana-pull-request/builds/328943) succeeded e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767
* :broken_heart: [Build #328889](https://buildkite.com/elastic/kibana-pull-request/builds/328889) failed cb11233c032a348b250a9885325e9059846d193c
* :broken_heart: [Build #328604](https://buildkite.com/elastic/kibana-pull-request/builds/328604) failed 49d57d569efe8b4449cdb209f8df87a95ce7be89
* :broken_heart: [Build #328550](https://buildkite.com/elastic/kibana-pull-request/builds/328550) failed 1f873a047add06af221e6bbfa3a87d6c1afe5eaf

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330216","number":330216,"commit":"ff25739b8a75df9c00b3e8b3fa21b265ef87a165"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328996","number":328996,"commit":"3d135c12d3ce43fdfc588d0d488fe650770062db"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328943","number":328943,"commit":"e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328889","number":328889,"commit":"cb11233c032a348b250a9885325e9059846d193c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328604","number":328604,"commit":"49d57d569efe8b4449cdb209f8df87a95ce7be89"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328550","number":328550,"commit":"1f873a047add06af221e6bbfa3a87d6c1afe5eaf"}],"number":330216}
buildkite-pr-comment-->

alexreal1314 committed:
fix graph types and add unit tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/330677)
* Commit: ed5ab19034925513fdb54b1949a24076ad39e74e

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/330677#0198c179-90b6-4c46-b384-bf70e5f50e76)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/330677#0198c179-90b6-4c46-b384-bf70e5f50e76) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/330677/jobs/0198c179-90b6-4c46-b384-bf70e5f50e76/artifacts/0198c18c-5016-402c-b2dd-d8b449984734) Scout: [ platform / streams_app ] plugin / stateful - Stream data mapping - schema editor - should allow unmapping a field

### History
* :green_heart: [Build #330216](https://buildkite.com/elastic/kibana-pull-request/builds/330216) succeeded ff25739b8a75df9c00b3e8b3fa21b265ef87a165
* :yellow_heart: [Build #328996](https://buildkite.com/elastic/kibana-pull-request/builds/328996) was flaky 3d135c12d3ce43fdfc588d0d488fe650770062db
* :green_heart: [Build #328943](https://buildkite.com/elastic/kibana-pull-request/builds/328943) succeeded e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767
* :broken_heart: [Build #328889](https://buildkite.com/elastic/kibana-pull-request/builds/328889) failed cb11233c032a348b250a9885325e9059846d193c
* :broken_heart: [Build #328604](https://buildkite.com/elastic/kibana-pull-request/builds/328604) failed 49d57d569efe8b4449cdb209f8df87a95ce7be89

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330677","number":330677,"commit":"ed5ab19034925513fdb54b1949a24076ad39e74e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330216","number":330216,"commit":"ff25739b8a75df9c00b3e8b3fa21b265ef87a165"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328996","number":328996,"commit":"3d135c12d3ce43fdfc588d0d488fe650770062db"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328943","number":328943,"commit":"e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328889","number":328889,"commit":"cb11233c032a348b250a9885325e9059846d193c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328604","number":328604,"commit":"49d57d569efe8b4449cdb209f8df87a95ce7be89"}],"number":330677}
buildkite-pr-comment-->

kfirpeled reviewed (commented):
Overall, looks solid. Nicely done @alexreal1314 !

Besides few comments we would need to address at some point the following:

1. When disabled, I'd expect the tooltip to describe why it is being disabled. To help the user troubleshoot. Work with @nick-alayil on the text

<img width="700" alt="Image" src="https://github.com/user-attachments/assets/69568ed2-e342-42c7-b553-614b93bf0032" />

2. The preview mode of the generic entity popover requires some polishing

- It misses the show full details button at its footer
- It is too simple to override the current left flyout - this is not aligned with other popovers in the security system which disable (sadly, we've missed a few there as well) that in preview mode

Attaching here a video for comparison

https://github.com/user-attachments/assets/7a836319-8654-4eb7-ae8f-6b37b3eb8efd


Let me know once you address the comments and I'll do another quick review for you
Thanks Alex!

alexreal1314 committed:
update preview mode of generic entity flyout

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334319)
* Commit: c33af23bece2182035bcb8ca0dce3f6833cc44d3

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/334319#0198fcd4-1c46-4da2-94f4-12009bd756bc)

### History
* :yellow_heart: [Build #330677](https://buildkite.com/elastic/kibana-pull-request/builds/330677) was flaky ed5ab19034925513fdb54b1949a24076ad39e74e
* :green_heart: [Build #330216](https://buildkite.com/elastic/kibana-pull-request/builds/330216) succeeded ff25739b8a75df9c00b3e8b3fa21b265ef87a165
* :yellow_heart: [Build #328996](https://buildkite.com/elastic/kibana-pull-request/builds/328996) was flaky 3d135c12d3ce43fdfc588d0d488fe650770062db
* :green_heart: [Build #328943](https://buildkite.com/elastic/kibana-pull-request/builds/328943) succeeded e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767
* :broken_heart: [Build #328889](https://buildkite.com/elastic/kibana-pull-request/builds/328889) failed cb11233c032a348b250a9885325e9059846d193c

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334319","number":334319,"commit":"c33af23bece2182035bcb8ca0dce3f6833cc44d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330677","number":330677,"commit":"ed5ab19034925513fdb54b1949a24076ad39e74e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330216","number":330216,"commit":"ff25739b8a75df9c00b3e8b3fa21b265ef87a165"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328996","number":328996,"commit":"3d135c12d3ce43fdfc588d0d488fe650770062db"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328943","number":328943,"commit":"e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328889","number":328889,"commit":"cb11233c032a348b250a9885325e9059846d193c"}],"number":334319}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334329)
* Commit: 334b07248c98c1cf3700f3182ea8d74156d16cf8
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-231370)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-231370-334b07248c98`

### History
* :broken_heart: [Build #334319](https://buildkite.com/elastic/kibana-pull-request/builds/334319) failed c33af23bece2182035bcb8ca0dce3f6833cc44d3
* :yellow_heart: [Build #330677](https://buildkite.com/elastic/kibana-pull-request/builds/330677) was flaky ed5ab19034925513fdb54b1949a24076ad39e74e
* :green_heart: [Build #330216](https://buildkite.com/elastic/kibana-pull-request/builds/330216) succeeded ff25739b8a75df9c00b3e8b3fa21b265ef87a165
* :yellow_heart: [Build #328996](https://buildkite.com/elastic/kibana-pull-request/builds/328996) was flaky 3d135c12d3ce43fdfc588d0d488fe650770062db
* :green_heart: [Build #328943](https://buildkite.com/elastic/kibana-pull-request/builds/328943) succeeded e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767
* :broken_heart: [Build #328889](https://buildkite.com/elastic/kibana-pull-request/builds/328889) failed cb11233c032a348b250a9885325e9059846d193c

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334329","number":334329,"commit":"334b07248c98c1cf3700f3182ea8d74156d16cf8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334319","number":334319,"commit":"c33af23bece2182035bcb8ca0dce3f6833cc44d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330677","number":330677,"commit":"ed5ab19034925513fdb54b1949a24076ad39e74e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330216","number":330216,"commit":"ff25739b8a75df9c00b3e8b3fa21b265ef87a165"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328996","number":328996,"commit":"3d135c12d3ce43fdfc588d0d488fe650770062db"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328943","number":328943,"commit":"e7a5754c0bd4b2ad79ffa952fda5b48dc7fa6767"}],"number":334329}
buildkite-pr-comment-->

alexreal1314 commented:
> Overall, looks solid. Nicely done @alexreal1314 !
> 
> Besides few comments we would need to address at some point the following:
> 
> 1. When disabled, I'd expect the tooltip to describe why it is being disabled. To help the user troubleshoot. Work with @nick-alayil on the text
> 
> <img alt="Image" width="700" src="https://private-user-images.githubusercontent.com/61654899/481385982-69568ed2-e342-42c7-b553-614b93bf0032.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY2NDIzNTEsIm5iZiI6MTc1NjY0MjA1MSwicGF0aCI6Ii82MTY1NDg5OS80ODEzODU5ODItNjk1NjhlZDItZTM0Mi00MmM3LWI1NTMtNjE0YjkzYmYwMDMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODMxVDEyMDczMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlZGQzOGM5NDhhYzBmOTI1ZGY3ZDI1M2Y4MzAwZDAzNTVmNGVlNzI0ZWYwZDM4MmZjOWUyNjRhZjU4ZjRlYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uOwuOQtrrNocoSq2QvJnpajTDWs2ZDkzXL8OilRh0hU">
> 2. The preview mode of the generic entity popover requires some polishing
> 
> * It misses the show full details button at its footer
> * It is too simple to override the current left flyout - this is not aligned with other popovers in the security system which disable (sadly, we've missed a few there as well) that in preview mode
> 
> Attaching here a video for comparison
> 
>  Screen.Recording.2025-08-24.at.17.31.31.mov 
> Let me know once you address the comments and I'll do another quick review for you Thanks Alex!

1. Hi @nick-alayil do you have a suggestion for a tooltip message? I was thinking that something like 'Details unavailable - this entity was no enriched'

2. @kfirpeled - I changed it to be similar to the 'event details' footer - with 'show full details' text and take action buttons when in preview mode.

alexreal1314 commented on x-pack/solutions/security/plugins/cloud_security_posture/server/routes/graph/parse_records.ts:
```diff
@@ -140,6 +154,7 @@ const createNodes = (records: GraphEdge[], context: Omit<ParseContext, 'edgesMap
     [...actorIdsArray, ...targetIdsArraySafe].forEach((id) => {
       if (nodesMap[id] === undefined) {
         nodesMap[id] = {
+          documentsData: getEntityDocuments(id, actorsDocDataArray, targetsDocDataArray),
```
@kfirpeled are you talking about a case where we can have a specific entity.id where in one log its under actor.entity.id and in another log its under target.entity.id field? because In that case we would end up with documentsData array with two items - first we got from actorsDocDataArray and the second from targetsDocDataArray.

do we want to split this entire forEach loop?

alexreal1314 committed:
optimize parsing nodes logic to avoid duplications of document data in the same node

alexreal1314 committed:
add tooltip when hovering on node list item with entity data

update storybook tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

alexreal1314 committed:
add data test sub id to node entity tooltip

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/336151)
* Commit: c8438d1cf16f2bc045f3c9e9c80e1484bb8af553
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-231370)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-231370-c8438d1cf16f`

### History
* :broken_heart: [Build #336064](https://buildkite.com/elastic/kibana-pull-request/builds/336064) failed 48415caa5e6c16c9607bd27cbbd4af93e93934bd
* :broken_heart: [Build #335915](https://buildkite.com/elastic/kibana-pull-request/builds/335915) failed f297db4fecd29bb93535471a735015378a17bffa
* :broken_heart: [Build #335355](https://buildkite.com/elastic/kibana-pull-request/builds/335355) failed b4e71b1a1a5afcf45df1ceab91df5ec57ddcb84e
* :broken_heart: [Build #335231](https://buildkite.com/elastic/kibana-pull-request/builds/335231) failed d624c8e233eddc6b2b10e85097e5c607cdb97a70
* :green_heart: [Build #334329](https://buildkite.com/elastic/kibana-pull-request/builds/334329) succeeded 334b07248c98c1cf3700f3182ea8d74156d16cf8

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336151","number":336151,"commit":"c8438d1cf16f2bc045f3c9e9c80e1484bb8af553"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336064","number":336064,"commit":"48415caa5e6c16c9607bd27cbbd4af93e93934bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335915","number":335915,"commit":"f297db4fecd29bb93535471a735015378a17bffa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335355","number":335355,"commit":"b4e71b1a1a5afcf45df1ceab91df5ec57ddcb84e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335231","number":335231,"commit":"d624c8e233eddc6b2b10e85097e5c607cdb97a70"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334329","number":334329,"commit":"334b07248c98c1cf3700f3182ea8d74156d16cf8"}],"number":336151}
buildkite-pr-comment-->

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/utils.ts:
```diff
@@ -76,14 +70,18 @@ export const getSingleDocumentData = (
     node.documentsData.find((doc) => doc.type === 'alert') ??
-    node.documentsData.find((doc) => doc.type === 'event');
+    node.documentsData.find((doc) => doc.type === 'event') ??
+    node.documentsData.find((doc) => doc.type === 'entity');
```
@kfirpeled fixed, I updated the function so now if mode === 'single-entity' it prioritized finding the entity document first.

alexreal1314 commented:
> > 1. Hi @nick-alayil do you have a suggestion for a tooltip message? I was thinking that something like 'Details unavailable - this entity was not enriched' could be appropriate.
> 
> My recommendation for now would be - `Details not available` and keep it simple. I changed from unavailable to not available since unavailable is often perceived as implying a temporary situation. In long run, we could use action oriented tooltips like `Details not available -- enrich this entity to view more.` with a hyperlink to entity store enablement/cloud asset discovery integration etc... For now, that would be an overkill. Cc @larissaoliveiraelastic

Thanks @nick-alayil here is the tooltip:
<img width="1904" height="927" alt="image" src="https://github.com/user-attachments/assets/8e8f5440-2a17-4246-bbcb-e01aae212e32" />

cc @kfirpeled

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/339081)
* Commit: 9200069fc721f6fbc7d399ab599701f9b96ca9b1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-231370)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-231370-9200069fc721`

### History
* :green_heart: [Build #338454](https://buildkite.com/elastic/kibana-pull-request/builds/338454) succeeded 2000565af9d3c70596b926bed1aaf8262ef9e000
* :broken_heart: [Build #336218](https://buildkite.com/elastic/kibana-pull-request/builds/336218) failed 29be2db4edc2dca98c0af4d69d0be0c7d80c28b5
* :green_heart: [Build #336151](https://buildkite.com/elastic/kibana-pull-request/builds/336151) succeeded c8438d1cf16f2bc045f3c9e9c80e1484bb8af553
* :broken_heart: [Build #336064](https://buildkite.com/elastic/kibana-pull-request/builds/336064) failed 48415caa5e6c16c9607bd27cbbd4af93e93934bd
* :broken_heart: [Build #335915](https://buildkite.com/elastic/kibana-pull-request/builds/335915) failed f297db4fecd29bb93535471a735015378a17bffa

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339081","number":339081,"commit":"9200069fc721f6fbc7d399ab599701f9b96ca9b1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338454","number":338454,"commit":"2000565af9d3c70596b926bed1aaf8262ef9e000"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336218","number":336218,"commit":"29be2db4edc2dca98c0af4d69d0be0c7d80c28b5"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336151","number":336151,"commit":"c8438d1cf16f2bc045f3c9e9c80e1484bb8af553"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336064","number":336064,"commit":"48415caa5e6c16c9607bd27cbbd4af93e93934bd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335915","number":335915,"commit":"f297db4fecd29bb93535471a735015378a17bffa"}],"number":339081}
buildkite-pr-comment-->

kfirpeled reviewed (approved):
Thank you for the patience and letting me review as well!
 I tested it on serverless, works very well!

Added few comments

alexreal1314 committed:
make small improvements to parseRecords function and fix menu popver icon css

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/340799)
* Commit: 1101f827af63cd2bb3607baa6d9f1020a31943dc
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-231370)
* [Cloud Deployment](https://kibana-pr-231370-a789ac.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #339081](https://buildkite.com/elastic/kibana-pull-request/builds/339081) was flaky 9200069fc721f6fbc7d399ab599701f9b96ca9b1
* :green_heart: [Build #338454](https://buildkite.com/elastic/kibana-pull-request/builds/338454) succeeded 2000565af9d3c70596b926bed1aaf8262ef9e000
* :broken_heart: [Build #336218](https://buildkite.com/elastic/kibana-pull-request/builds/336218) failed 29be2db4edc2dca98c0af4d69d0be0c7d80c28b5
* :green_heart: [Build #336151](https://buildkite.com/elastic/kibana-pull-request/builds/336151) succeeded c8438d1cf16f2bc045f3c9e9c80e1484bb8af553
* :broken_heart: [Build #336064](https://buildkite.com/elastic/kibana-pull-request/builds/336064) failed 48415caa5e6c16c9607bd27cbbd4af93e93934bd

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/340799","number":340799,"commit":"1101f827af63cd2bb3607baa6d9f1020a31943dc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/339081","number":339081,"commit":"9200069fc721f6fbc7d399ab599701f9b96ca9b1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338454","number":338454,"commit":"2000565af9d3c70596b926bed1aaf8262ef9e000"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336218","number":336218,"commit":"29be2db4edc2dca98c0af4d69d0be0c7d80c28b5"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336151","number":336151,"commit":"c8438d1cf16f2bc045f3c9e9c80e1484bb8af553"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336064","number":336064,"commit":"48415caa5e6c16c9607bd27cbbd4af93e93934bd"}],"number":340799}
buildkite-pr-comment-->
